### ISSUE TITLE
The SDL2 Library is not working on linux

### ISSUE THREAD

**Issue by TheRawMeatball** [beefytech/Beef#202]
Does anyone know anything about this?

When I took the source code for the demo and tried compiling it on my linux machine, this error came up.

```
ERROR: Type could not be found (are you missing a using directive or library reference?) at line 5128:11 in /mnt/33CA263211FBF90F/Beef-master/BeefLibs/SDL2/src/SDL2.bf
   public Windows.HWnd window; // Refers to an HWND
          ^^^^^^^
ERROR: Type could not be found (are you missing a using directive or library reference?) at line 5129:11 in /mnt/33CA263211FBF90F/Beef-master/BeefLibs/SDL2/src/SDL2.bf
   public Windows.Handle hdc; // Refers to an HDC
          ^^^^^^^
Compile failed.
```

Is the problem in the demo or is the SDL library stuck to Windows?

**Comment by bfiete** [comment]
That particular compiler bug is fixed at https://github.com/beefytech/Beef/commit/7701485fd4bb70afc5b645fddcd926b3d4b779a2. The sample will compile but you'll get linker errors because the linker flags have not been set up, but presumably that sample should work on Linux unless there's other bugs preventing it from doing so.

Anyone want to try?

**Comment by TheRawMeatball** [comment]
Nope, doesn't seem to work. I got the same error message I posted on the thread for the pull request.
I'll post it here as well for completeness sake:
```
Call parameter type does not match function signature!
i8* %0
 i32  %4 = call i32 @BfpSystem_SetCommandLine(i8* %0, i8* %1)
in function WinMain
LLVM ERROR: Broken function found, compilation aborted!
```

**Comment by bfiete** [comment]
You've built the latest release from source?

**Comment by bfiete** [comment]
As in, the trunk from today?

**Comment by TheRawMeatball** [comment]
No, I just replaced the SDL2.bf file with the latest version. Also, I've done some digging and found out a bit more: 

The  `Call parameter type does not match function signature!` error is coming from BfCompiler.cpp (lines 1638 - 1676) where the `#ifndef BF_PLATFORM_WINDOWS` block doesn't account for being called with `TargetType == BfTargetType_BeefWindowsApplication`,

**Comment by TheRawMeatball** [comment]
I think the root cause of all these problems is that the TargetType enum in the core compiler looks like this:
```
enum BfTargetType
{
	BfTargetType_BeefConsoleApplication,
	BfTargetType_BeefWindowsApplication,		
	BfTargetType_BeefLib,
	BfTargetType_BeefDynLib,
	BfTargetType_CustomBuild,
	BfTargetType_C_ConsoleApplication,
	BfTargetType_C_WindowsApplication,
	BfTargetType_BeefTest,
	BfTargetType_BeefApplication_StaticLib,
	BfTargetType_BeefApplication_DynamicLib
};
```
Because BeefGUIApplication turns into BeefWindowsApplication, all Linux GUI projects get misconfigured, and in the case of the `Call parameter type does not match function signature!` error cause an internal conflict between the `TargetType` enum and `BF_PLATFORM_WINDOWS` directive.

Tl;dr: I dont think this will get patched out with just a couple preprocesor directives.

**Comment by bfiete** [comment]
You're right. Fix at https://github.com/beefytech/Beef/commit/1f2632786c4efbe236281cf5bc0446450b89ceea.
Also updated the sample project at https://github.com/beefytech/Beef_website/commit/bb5b4a414ffed3371ae98d0266b891119f8ae49d.

I've actually compiled and tested SpaceGame on Linux now, through WSL/Ubuntu.
